iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/PackageInstallationCacheCheck.java

```diff
@@ -148,6 +176,46 @@ private static List<CommandDetector.Command> detectRemoveCacheCommands(
       .toList();
   }
 
+  /* Compute a set of commands for which their cache path is present is the provided list. */
+  private static Set<String> computeCachedCommands(List<String> cachedPaths) {
+    return cachedPaths.stream()
+      .flatMap(cachedPath -> CACHE_TO_COMMANDS.getOrDefault(cachedPath, Collections.emptySet()).stream())
+      .collect(Collectors.toSet());
+  }
+
+  /**
+   * Retrieve all mounted flags that has a cache type and return their target.
+   */
+  private static List<String> retrieveMountedCachePath(RunInstruction runInstruction) {
+    return runInstruction.options().stream()
+      .filter(option -> "mount".equals(option.name()))
+      .map(option -> ArgumentResolution.of(option.value()))
+      .filter(ArgumentResolution::isResolved)
+      .map(ArgumentResolution::value)
+      .map(PackageInstallationCacheCheck::computeMapMountDetails)
+      .filter(entry -> "cache".equals(entry.get("type")))
+      .map(entry -> entry.get("target"))
+      .filter(Objects::nonNull)
```

Comment:
It can be null in the case of `--mount=type=cache`, because we change it into a map {"type": "cache"}, and from it we look for a key "target".
The `entry` variable name is misleading, I believe I put that name because I went through a `map.entrySet()` in a previous version, but now I changed it to `map` instead.